libs/natparse/src/main/java/org/amshove/natparse/lexing/Lexer.java

```diff
@@ -891,6 +913,53 @@ private boolean mightBeDataFormat(String possibleDataFormat)
 		return floatingPointCount < 2;
 	}
 
+	private boolean consumeNaturalHeader()
+	{
+		if (isAtLineStart() && (isSingleAsteriskComment() || isInlineComment()))
+		{
+			//NOOP
+		}
+		else
+			return false;
+
+		scanner.start();
+		while (!isLineEnd() && !scanner.isAtEnd())
+		{
+			scanner.advance();
+		}
+		String s = scanner.lexemeText().stripTrailing();
+
+		if (inHdr)
+		{
+			if (s.contains("* <Natural Source Header"))
+			{
+				sourceHeader = new NaturalHeader(mode, lineIncrement);
+				hdrDone = true;
+			}
+			else
+			{
+				if (s.contains("* :Mode"))
+				{
+					mode = NaturalProgrammingMode.fromString(s.substring(s.length() - 1));
+				}
+				else
+					if (s.contains("* :LineIncrement"))
+					{
+						s = s.replaceAll("[^0-9]+", "");
```

Comment:
[According to SonarQube](https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=102&id=MarkusAmshove_natls&open=AYWHLM_s0zYNSmsEaCDJ) the `[^0-9]+` can be replaced with `\\D`